src/core/replication_rules.rs

```diff
@@ -1,245 +1,394 @@
+```
+use std::io::Cursor;
 
-/// Stores meta information about replicated component.
-#[derive(Clone)]
-pub(crate) struct ReplicationInfo {
-    /// Function that serializes component into bytes.
-    pub(crate) serialize: SerializeFn,
+use bevy::{prelude::*, ptr::Ptr};
+use bevy_replicon::{
+    client::client_mapper::ServerEntityMap,
+    core::{
+        replication_rules::{self, GroupReplication, ReplicationRule},
+        replication_fns::{self, ReplicationFns, ComponentFns},
+        replicon_tick::RepliconTick,
+    },
+    prelude::*,
+};
+use serde::{Deserialize, Serialize};
 
-    /// Function that deserializes component from bytes and inserts it to [`EntityWorldMut`].
-    pub(crate) deserialize: DeserializeFn,
+# let mut app = App::new();
+# app.add_plugins(RepliconPlugins);
+app.replicate_group::<PlayerBundle>();
 
-    /// Function that removes specific component from [`EntityWorldMut`].
-    pub(crate) remove: RemoveComponentFn,
+#[derive(Bundle)]
+struct PlayerBundle {
+    transform: Transform,
+    player: Player,
+    replication: Replication,
 }
 
-/// Marks entity for replication.
-#[derive(Component, Clone, Copy, Default, Reflect, Debug)]
-#[reflect(Component)]
-pub struct Replication;
-
-/// Same as [`ComponentId`], but consistent between server and clients.
-///
-/// Internally represents index of [`ReplicationInfo`].
-#[derive(Clone, Copy, Deserialize, Eq, Hash, PartialEq, Serialize)]
-pub(crate) struct ReplicationId(usize);
-
-/// Default serialization function.
-pub fn serialize_component<C: Component + Serialize>(
-    component: Ptr,
-    cursor: &mut Cursor<Vec<u8>>,
-) -> bincode::Result<()> {
-    // SAFETY: Function called for registered `ComponentId`.
-    let component: &C = unsafe { component.deref() };
-    DefaultOptions::new().serialize_into(cursor, component)
+#[derive(Component, Deserialize, Serialize)]
+struct Player;
+
+impl GroupReplication for PlayerBundle {
+    fn register(world: &mut World, replication_fns: &mut ReplicationFns) -> ReplicationRule {
+        // Customize serlialization to serialize only `translation`.
+        let transform_id = world.init_component::<Transform>();
+        let transform_fns_id = replication_fns.register_component_fns(ComponentFns {
+            // For function definitions see the example from `AppReplicationExt::replicate_with`.
+            serialize: serialize_translation,
+            deserialize: deserialize_translation,
+            remove: replication_fns::remove::<Transform>, // Use default removal function.
+        });
+
+        // Serialize `player` as usual.
+        let visibility_id = world.init_component::<Player>();
+        let visibility_fns_id = replication_fns.register_component_fns(ComponentFns {
+            serialize: replication_fns::serialize::<Player>,
+            deserialize: replication_fns::deserialize::<Player>,
+            remove: replication_fns::remove::<Player>,
+        });
```

Comment:
Good call! I thought about this too, but didn't come up with a name and forgot to come back to it :D

Reply:
I created `ComponentFns::<T>::default_fns()` and `ComponentFns::<T>::default_fns_mapped()`, what do you think about these names?
I wanted to emphasis on "default" since it's how this functions documented in docs.